checker.go

```diff
@@ -166,14 +166,14 @@ type errorMatchesChecker struct {
 // Check implements Checker.Check by checking that got is an error whose
 // Error() matches args[0].
 func (c *errorMatchesChecker) Check(got interface{}, args []interface{}, note func(key string, value interface{})) error {
+	if got == nil {
+		return errors.New("no error found")
```

Comment:
Maybe `want an error, got nil` or something similar?
I'd like to avoid the `error: no error found` in the output, which can be a bit confusing.